src/identities/mnemonic.ts

```diff
@@ -1,30 +1,177 @@
+    if (args.value.language) {
+      bip39.setDefaultWordlist(args.value.language);
+    } else {
+      bip39.setDefaultWordlist('english');
+    }
+
+    // validate the mnemonic
+    if (!bip39.validateMnemonic(args.value.mnemonic)) {
+      throw new Error('Mnemonic is not valid.');
     }
 
-    this.words = (args.value as string).split(' ');
-    console.log(this.words);
+    // set up the wallet's seed.
+    this.walletSeed = bip39.mnemonicToSeedSync(
+      args.value.mnemonic,
+      args.value.passphrase
+    );
+    // generate the master private key from the wallet seed
+    this.masterPrivateKeyNode = bip32fromSeed(this.walletSeed, this.network);
+    // generate the master blinding key from the seed
+    this.masterBlindingKeyNode = slip77fromSeed(this.walletSeed);
   }
 
-  signPset(psetBase64: string): string {
-    console.log(psetBase64);
-    return '';
+  private getNextKeypair(): { publicKey: Buffer; privateKey: Buffer } {
+    const baseNode = this.masterPrivateKeyNode.derivePath(this.derivationPath);
+    const wif: string = baseNode.deriveHardened(this.index).toWIF();
+    const { publicKey, privateKey } = ECPair.fromWIF(wif, this.network);
+    this.index += 1;
+    return { publicKey: publicKey!, privateKey: privateKey! };
+  }
+
+  private getBlindingKeyPair(
+    scriptPubKey: Buffer
+  ): { publicKey: Buffer; privateKey: Buffer } {
+    const { publicKey, privateKey } = this.masterBlindingKeyNode.derive(
+      scriptPubKey
+    );
+    return { publicKey: publicKey!, privateKey: privateKey! };
+  }
+
+  getNextConfidentialAddress(): AddressInterfaceExtended {
+    const currentIndex = this.index;
+    // get the next key pair
+    const keyPair = this.getNextKeypair();
+    // use the public key to compute the scriptPubKey
+    const script: Buffer = payments.p2wpkh({
+      pubkey: keyPair.publicKey!,
+      network: this.network,
+    }).output!;
+    // generate the blindKeyPair from the scriptPubKey
+    const blindingKeyPair = this.getBlindingKeyPair(script);
+    // with blindingPublicKey & signingPublicKey, generate the confidential address
+    const { confidentialAddress } = payments.p2wpkh({
+      pubkey: keyPair.publicKey!,
+      blindkey: blindingKeyPair.publicKey!,
+      network: this.network,
+    });
+    // create the address generation object
+    const newAddressGeneration: AddressInterfaceExtended = {
+      address: {
+        confidentialAddress: confidentialAddress!,
+        blindingPrivateKey: blindingKeyPair.privateKey!.toString('hex'),
+      },
+      derivationPath: `${this.derivationPath}/${currentIndex}`,
+      signingPrivateKey: keyPair.privateKey!.toString('hex'),
+      scriptPubKey: script,
+    };
+    // store the generation inside local cache
+    this.confidentialAddresses.push(newAddressGeneration);
+    // return the generation data
+    return newAddressGeneration;
+  }
+
+  unblindUtxo(blindedUtxo: Output): UnblindOutputResult {
```

Comment:
Why do you think is necessary? It is already implemented somewhere else (ie utils) and we could always get the blinding key either from cache or other getters.

Reply:
I agree, I'll remove the method. maybe we need to add a `getBlindingKey(script | address)` to `IdentityInterface`. The wallet will probably need a standard way to unblind outputs / dump blinding keys?

Reply:
PS: what exactly is the difference between a change address and a not-for-change one?  :/

Reply:
ACK for getBlindingKey